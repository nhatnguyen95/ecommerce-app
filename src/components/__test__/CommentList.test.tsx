import 'react-native';
import React from 'react';
import { render } from '@testing-library/react-native';
import { it } from '@jest/globals';

import CommentList from '../CommentList';
import { Provider } from 'react-redux';
import { mockStore } from '../../../__mocks__/redux-store';
import { NavigationContainer } from '@react-navigation/native';

function setup() {
  const component = render(
    <Provider store={mockStore}>
      <NavigationContainer>
        <CommentList parrentCommentId={1} />
      </NavigationContainer>
    </Provider>
  );
  return component;
}

describe('Testing CommentList component', () => {
  it('should render correctly', () => {
    const tree = setup().toJSON();
    expect(tree).toMatchSnapshot()
  });

})

