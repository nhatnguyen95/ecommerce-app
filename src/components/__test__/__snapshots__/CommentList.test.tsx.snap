// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "flex-start",
        "fontStyle": "italic",
        "fontWeight": "bold",
        "marginVertical": 16,
      }
    }
  />
  <Text />
  <Text
    style={
      {
        "alignSelf": "flex-end",
        "fontStyle": "italic",
      }
    }
  >
    Replies (
    0
    )
  </Text>
  <View
    style={
      {
        "backgroundColor": "#f9f9f9",
        "marginHorizontal": 16,
      }
    }
  />
</View>
`;
